### PR TITLE
SEARCH-433    CMIS QL - Support inner and left outer joins from/to a …

### PR DESCRIPTION
**Pull request by andyhind**
Left outer join between a single type and any number of aspects
- Data model and parser changes

### CODE REVIEW

File: src/main/java/org/alfresco/repo/search/impl/querymodel/impl/BaseSelector.java [hunk 1]
```diff
@@ -44,39 +45,50 @@
...
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.alfresco.repo.search.impl.querymodel.Selector#getType()
-     */
+    
+    
+    @Override
+	public JoinType getJoinType()
```
Comment (michaelsuzukisagi) [inline src/main/java/org/alfresco/repo/search/impl/querymodel/impl/BaseSelector.java]: Formatting

File: src/main/java/org/alfresco/repo/search/impl/querymodel/Source.java [hunk 2]
```diff
@@ -40,4 +40,8 @@
     public Selector getSelector(String name);
     
     public List<Set<String>> getSelectorGroups(FunctionEvaluationContext functionContext);
+
+    public JoinType getJoinType();
+    
+	public void setJoinType(JoinType joinType);
```
Comment (michaelsuzukisagi) [inline src/main/java/org/alfresco/repo/search/impl/querymodel/Source.java]: formatting

File: src/test/resources/org/alfresco/repo/search/impl/parsers/cmis_test.gunit [hunk 3]
```diff
@@ -17,6 +17,15 @@ query:
 "SELECT cmis:name,, cmis:typeId FROM cmis:document" FAIL
 "SELECT * FROM *" FAIL
```
Comment (michaelsuzukisagi) [inline src/test/resources/org/alfresco/repo/search/impl/parsers/cmis_test.gunit:19]: I would add a test to right outer join and expect it to throw an error

File: src/main/java/org/alfresco/repo/search/impl/querymodel/impl/BaseSelector.java (hunk 1 above)
Comment (andyhind) [reply src/main/java/org/alfresco/repo/search/impl/querymodel/impl/BaseSelector.java]: Done

File: src/main/java/org/alfresco/repo/search/impl/querymodel/Source.java (hunk 2 above)
Comment (andyhind) [reply src/main/java/org/alfresco/repo/search/impl/querymodel/Source.java]: Done

File: src/test/resources/org/alfresco/repo/search/impl/parsers/cmis_test.gunit (hunk 3 above)
Comment (andyhind) [reply src/test/resources/org/alfresco/repo/search/impl/parsers/cmis_test.gunit:19]: I have added simple test for ROJ fail.

**Pull request merged**